src/zarr/core/buffer/gpu.py

```diff
@@ -59,7 +59,7 @@ def __init__(self, array_like: ArrayLike) -> None:
 
         if array_like.ndim != 1:
             raise ValueError("array_like: only 1-dim allowed")
-        if array_like.dtype != np.dtype("B"):
+        if array_like.dtype.itemsize != 1:
```

Comment:
The new tests in `test_nvcomp.py` were failing without this change.

I'd like to get us to a point where we don't care as much about the details of the buffer passed in here. This is an OK start I think.

Reply:
what exactly does this check for? It's not clear to me why any numpy array that can be viewed as bytes wouldn't be allowed in here

Reply:
i'm not even convinced that we need `Buffer` / `NDBuffer`, when `Buffer` is just a special case of `NDBuffer` where there's 1 dimension and the data type is bytes

Reply:
we could even express this formally by:
- making `NDBuffer` generic with two type parameters (number of dimensions and dtype)
- having APIs that insist on consuming a `Buffer` instead insist on consuming `NDBuffer[Literal[1], np.uint8]`

Reply:
> It's not clear to me why any numpy array that can be viewed as bytes wouldn't be allowed in here

I think this is mainly because `NDBuffer` objects don't need to be contiguous, but `Buffer` objects must be contiguous in memory which might be important when we send those out to codecs that expect a contiguous memory slice.

But I agree that we can probably merge those two and make `Buffer` a specialization of `NDBuffer`.